examples/cxx/event_based_communication/src/subscriber.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#include <iostream>
+
+#include "iox/duration.hpp"
+#include "iox/into.hpp"
+#include "iox2/file_descriptor.hpp"
+#include "iox2/listener.hpp"
+#include "iox2/node.hpp"
+#include "iox2/notifier.hpp"
+#include "iox2/service_name.hpp"
+#include "iox2/service_type.hpp"
+#include "iox2/subscriber.hpp"
+#include "iox2/waitset.hpp"
+#include "pubsub_event.hpp"
+#include "transmission_data.hpp"
+
+constexpr iox::units::Duration DEADLINE = iox::units::Duration::fromSeconds(2);
+
+using namespace iox2;
+
+// High-level subscriber class that contains besides a subscriber also a notifier
+// and a listener. The notifier is used to send events like
+// `PubSubEvent::ReceivedSample` or to notify the publisher that a new subscriber
+// connected.
+// The listener waits for events originating from the publisher like
+// `PubSubEvent::SentSample`.
+class EventBasedSubscriber : public FileDescriptorBased {
```

Comment:
Please also move this into an `hpp` file.

Reply:
I tried it, but I do not like it. I really prefer it when the essence is in one file in the case of an example.

The TransmissionData and the PubSubEvent are just simple helpers but for me, it is important that the user sees that they can define there own advanced publisher/subscriber, combine them with events and attach it to a waitset.